TelemetryEvents.swift

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+private let PrefKeySearches = "Telemetry.Searches"
+
+public class SearchTelemetryFactory {
```

Comment:
nit: This is more like a `SearchTelemetryEventFactory`. fwiw, nalexander once yelled at me for using `Factory` (granted, in Java) in the name of a class full of factory methods, as opposed to the proper factory design pattern. ymmv.